pkg/api/validation/validation.go

```diff
@@ -30,12 +30,21 @@ import (
 	"github.com/golang/glog"
 )
 
+const QualifiedNameErrorMsg string = " have to match [" + util.Dns1123SubdomainFmt + " / ] " + util.Dns1123LabelFmt + " pattern"
```

Comment:
1) done
2) ack
3.1) done
3.2) done
3.3) done
3.4) Maybe, but that's not what the code does - 0 is allowed for host port even though normally it's not.
4) ack
5) You mean to change this to functions like "AddXXXErrorDetail(foo)" to util/validation? SGTM, but I don't know how the code is structured in general. Let me know if I should do this.